couchbase/json-buckets.yml

```diff
@@ -0,0 +1,1071 @@
+        type: double_counter
+        label: cas misses
+        description: Compare and Swap misses
+
+      - name: cmd.get
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.cmd_get[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:sum()
+        type: double_counter
+        label: cmd gets
+        description: Number of get commands
+
+      - name: cmd.set
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.cmd_set[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:sum()
+        type: double_counter
+        label: cmd sets
+        description: Number of set commands
+
+      - name: docs.disk.actual.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_actual_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc actual disk size
+        unit: bytes
+        description: Couch docs total size on disk in bytes.
+
+      - name: docs.data.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_data_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc data disk size
+        unit: bytes
+        description: Couch docs data size in bytes
+
+      - name: docs.disk.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: docs disk size
+        unit: bytes
+        description: Couch docs total size in bytes
+
+      - name: docs.fragmentation
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_fragmentation[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc fragmentation
+        unit: "%"
+        description: Couch docs fragmentation
+
+      - name: data.spatial.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_data_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
+        label: spatial data size
+        unit: bytesals
+        description: Size of object data for spatial views
+
+      - name: disk.spatial.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
+        label: spatial disk size
+        unit: bytes
+        description: Amount of disk space occupied by spatial views
+
+      - name: ops.spatial
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_ops[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
```

Comment:
Counter? If we are reporting average number of something, that should be reflected in metric name, label, description. But in cases like this one we want to produce a counter. About a week ago we discussed that metrics like this one, if they report in some array element (which represents 1 second period) avg value of say X ops/sec, then in that second there were X ops which means we can just use function `sum()` on the array instead of `avg()` and return a count (in which case type would also be counter). From what I see there are many similar cases below, I will not comment on all of them, but please check them out.

Reply:
i did that for all metrics i know that are per second.
From this source
https://docs.epoch.nutanix.com/integrations/couchbase/
we can know which one one is per second metrics.

couch_spatial_ops unit is operations, not per second so i don't change to sum()

Reply:
Ok, does it report as a cumulative? If yes, I think we would need as_counter prefix and use just 0th element of array. If not, it probably reports number of operations for each second, so in that case we'd need sum() instead of avg().

Reply:
No metrics in this api is cumulative, if it is originally cumulative, they converted to per second.

Reply:
Ok, then it seems we have to use sum().

Reply:
I only use sum() for per second metrics, this metric unit is 'operation' only.

Reply:
That logic doesn't sound right. If it is count of operations that is reported for each second, that is even more reason to use sum(). Using avg makes no sense in this context.

Reply:
If it is number of operation for each second, it's unit should be operations/second. but in this case it's unit is only operation, so i guest it only average operation.